tool/tbot/proxy.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"path/filepath"
+
+	"github.com/gravitational/teleport/tool/tbot/config"
+	"github.com/gravitational/teleport/tool/tbot/tshwrap"
+	"github.com/gravitational/trace"
+)
+
+func onProxyCommand(botConfig *config.BotConfig, cf *config.CLIConf) error {
+	runner, err := tshwrap.NewRunner()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if err := tshwrap.CheckTSHSupported(runner); err != nil {
+		return trace.Wrap(err)
+	}
+
+	destination, err := tshwrap.GetDestination(botConfig, cf)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	destinationPath, err := tshwrap.GetDestinationPath(destination)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	identityTemplate, err := tshwrap.GetIdentityTemplate(destination)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	env, err := tshwrap.GetEnvForTSH(destination)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	identityPath := filepath.Join(destinationPath, identityTemplate.FileName)
```

Comment:
I do like moving to a simple `Wrapper` struct returned by `New()`. I'm wary of coupling all the extra fields so tightly because ultimately beyond just the path to tsh itself, all arguments to tsh are determined by the caller. The cost of gathering everything every time even if we don't need it is minimal, but then so is extracting just what we need given there's already helper functions.

Ultimately the difference feels pretty arbitrary and I'd rather err on the side of leaving things that aren't broken.